src/store.ts

```diff
@@ -453,6 +458,83 @@ export default function Store(
 
         mpInstance._Persistence.savePersistence(this.persistenceData);
     };
+
+    this.getFirstSeenTime = (mpid: MPID) => {
+        if (!mpid) {
+            return null;
+        }
+
+        if (
+            this.persistenceData &&
+            this.persistenceData[mpid] &&
+            this.persistenceData[mpid].fst
+        ) {
+            return this.persistenceData[mpid].fst;
+        } else {
+            return null;
+        }
+    };
+
+    this.setFirstSeenTime = (mpid: MPID, _time?: number) => {
+        if (!mpid) {
+            return;
+        }
+
+        const time = _time || new Date().getTime();
+
+        if (this.persistenceData) {
+            if (!this.persistenceData[mpid]) {
+                this.persistenceData[mpid] = {};
+            }
+            if (!this.persistenceData[mpid].fst) {
+                this.persistenceData[mpid].fst = time;
+                mpInstance._Persistence.savePersistence(this.persistenceData);
+            }
+        }
+    };
+
+    this.getLastSeenTime = (mpid: MPID) => {
+        if (!mpid) {
+            return null;
+        }
+
+        // TODO: We should store the current user in the Store's mpid rather than
```

Comment:
I like this Idea - it is probably done the other way as a "just in case", but if we can ensure it is also always on the store, then this could simply things.

Reply:
Yeah, I think I should create a ticket for this so we can decouple Identity from the Store as a next pass.